WcaOnRails/spec/models/competition_spec.rb

```diff
@@ -243,4 +243,52 @@
       expect(competition).not_to be_valid
     end
   end
+
+  describe "receive_registration_emails" do
+    let(:competition) { FactoryGirl.create :competition }
+    let(:delegate) { FactoryGirl.create :delegate }
+
+    it "computes receiving_registration_emails? via OR" do
+      expect(competition.receiving_registration_emails?(delegate.id)).to eq false
+
+      competition.delegates << delegate
+      expect(competition.receiving_registration_emails?(delegate.id)).to eq true
+
+      cd = competition.competition_delegates.find_by_delegate_id(delegate.id)
+      cd.update_column(:receive_registration_emails, false)
+      expect(competition.receiving_registration_emails?(delegate.id)).to eq false
+
+      competition.organizers << delegate
+      expect(competition.receiving_registration_emails?(delegate.id)).to eq true
+
+      co = competition.competition_organizers.find_by_organizer_id(delegate.id)
+      co.update_column(:receive_registration_emails, false)
+      expect(competition.receiving_registration_emails?(delegate.id)).to eq false
+    end
+
+    it "setting receive_registration_emails" do
+      competition.delegates << delegate
+      competition.save!
```

Comment:
AFAIK `competition.delegates << delegate` already triggers an `INSERT` (which is pretty unintuitive...).